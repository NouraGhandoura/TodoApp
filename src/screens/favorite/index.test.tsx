import React from 'react';
import { render, RenderAPI } from '@testing-library/react-native';
import Favorite from './index'
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);
const initialState = {
  todos: []
};

const store = mockStore(initialState);

describe('Favorite Screen', () => {

    const renderWithStore = (ui: React.ReactElement): RenderAPI => {
      return render(<Provider store={store}>{ui}</Provider>);
    };

      test('renders correctly and matches snapshot', () => {
      const { toJSON } = renderWithStore(<Favorite />);
      expect(toJSON()).toMatchSnapshot();
  });
});
