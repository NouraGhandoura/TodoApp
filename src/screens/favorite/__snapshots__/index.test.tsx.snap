// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite Screen renders correctly and matches snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#E4CBCB",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#E4CBCB",
        "flex": 1,
        "paddingHorizontal": 20,
      }
    }
  >
    <RCTScrollView
      style={
        {
          "backgroundColor": "#E4CBCB",
        }
      }
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 18,
                "marginTop": 200,
              }
            }
          >
            No favorite tasks available
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
